### PR TITLE
Fix error when save a layout and then come back to content the html i…

### PR DESCRIPTION
**Pull request by freddyucv**
…s not refrest with the new layout

### CODE REVIEW

File: src/app/api/services/dot-render-html/dot-render-html.service.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export class DotRenderHTMLService {
         return this.get(url, PageMode.LIVE, viewAsConfig);
     }
 
-    private get(url: string, pageMode: PageMode, viewAsConfig?: DotEditPageViewAs): Observable<DotRenderedPage> {
+    public get(url: string, pageMode: PageMode, viewAsConfig?: DotEditPageViewAs): Observable<DotRenderedPage> {
```
Comment (fmontes) [inline src/app/api/services/dot-render-html/dot-render-html.service.ts]: We talked about this method receives an object.

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout-advanced/dot-edit-layout-advanced.component.spec.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ describe('DotEditLayoutAdvancedComponent - Basic', () => {
         fixture = DOTTestBed.createComponent(DotEditLayoutAdvancedComponent);
         component = fixture.componentInstance;
         de = fixture.debugElement;
-        component.templateInode = '456';
+        //component.templateInode = '456';
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout-advanced/dot-edit-layout-advanced.component.spec.ts]: ?

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<dot-edit-layout-advanced [templateInode]="(pageState | async)?.template.inode" *ngIf="(pageState | async)?.template && isAdvancedTemplate | async else designer"></dot-edit-layout-advanced>
+<dot-edit-layout-advanced [pageState]="pageState | async" *ngIf="(pageState | async)?.template && isAdvancedTemplate | async else designer"></dot-edit-layout-advanced>
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.html]: Okey... this makes sense.

File: src/app/portlets/dot-edit-page/shared/models/dot-rendered-page-state.model.ts [hunk 4]
```diff
@@ -65,6 +65,17 @@ export class DotRenderedPageState {
         return this._viewAs;
     }
 
+    set(dotRenderedPageState: DotRenderedPageState): void {
+        this._canCreateTemplate = dotRenderedPageState.canCreateTemplate;
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/shared/models/dot-rendered-page-state.model.ts]: I think there is a TS pattern to do this better... but I don't know it :)

File: src/app/api/services/dot-render-html/dot-render-html.service.ts [hunk 5]
```diff
@@ -85,3 +103,9 @@ export class DotRenderHTMLService {
         return pageModeString[pageMode];
     }
 }
+
+class PageOptions {
```
Comment (fmontes) [inline src/app/api/services/dot-render-html/dot-render-html.service.ts]: DotPageOptions?

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.spec.ts [hunk 6]
```diff
@@ -101,8 +107,11 @@ describe('DotEditLayoutComponent with Edit Advanced Layout', () => {
         expect(layoutDesigner).toBeTruthy();
     });
 
-    it('should pass templateInode to the dot-edit-layout-advanced', () => {
+    fit('should pass pageState to the dot-edit-layout-advanced', () => {
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.spec.ts]: fit?

File: src/app/test/dot-page-state.service.mock.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    page: mockDotPage,
+    html: '',
+    layout: mockDotLayout,
+    canCreateTemplate: true,
+    viewAs: null
+};
+
+@Injectable()
+export class DotPageStateServiceMock {
+    get(url: string): Observable<DotRenderedPageState> {
```
Comment (fmontes) [inline src/app/test/dot-page-state.service.mock.ts]: But the original get receives an object, right?

File: src/app/api/services/dot-render-html/dot-render-html.service.ts [hunk 8]
```diff
@@ -36,7 +42,13 @@ export class DotRenderHTMLService {
      * @memberof DotRenderHTMLService
      */
     getPreview(url: string, viewAsConfig?: DotEditPageViewAs): Observable<DotRenderedPage> {
-        return this.get(url, PageMode.PREVIEW, viewAsConfig);
+        return this.get(
```
Comment (fmontes) [inline src/app/api/services/dot-render-html/dot-render-html.service.ts]: No new tests for this?

File: src/app/api/services/dot-render-html/dot-render-html.service.ts (hunk 5 above)
Comment (fmontes) [reply src/app/api/services/dot-render-html/dot-render-html.service.ts]: Why class and no interface?

File: src/app/api/services/dot-render-html/dot-render-html.service.ts (hunk 8 above)
Comment (freddyucv) [reply src/app/api/services/dot-render-html/dot-render-html.service.ts]: I missed it

File: src/app/api/services/dot-render-html/dot-render-html.service.ts (hunk 5 above)
Comment (freddyucv) [reply src/app/api/services/dot-render-html/dot-render-html.service.ts]: very much time coding just in java hahahahahahha

File: src/app/test/dot-page-state.service.mock.ts (hunk 7 above)
Comment (freddyucv) [reply src/app/test/dot-page-state.service.mock.ts]: no really

https://github.com/dotCMS/core-web/blob/master/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts#L45

**Pull request merged**